python/ray/train/v2/_internal/callbacks/placement_group_callback.py

```diff
@@ -0,0 +1,157 @@
+                )
+                self._stop_cleaner()
+                return
+
+            logger.debug(
+                f"PlacementGroupCleaner launched for run_id={train_run_context.run_id}"
+            )
+        except Exception as e:
+            logger.warning(
+                f"Failed to launch PlacementGroupCleaner: {e}. "
+                "Placement groups may not be cleaned up if controller exits ungracefully."
+            )
+            self._cleaner = None
+
+    def after_worker_group_start(self, worker_group: "WorkerGroup"):
+        """Register the worker group's placement group with the cleaner.
+
+        This is called after a worker group is successfully started.
+        """
+        if not self._cleaner:
+            logger.debug("No PlacementGroupCleaner to register placement group with")
+            return
+
+        try:
+            worker_group_state = worker_group.get_worker_group_state()
+            placement_group = worker_group_state.placement_group
+
+            # Placement group for worker group is created after worker group start.
+            # Only one placement group for worker group exists at a time.
+            if not self._controller_actor_id:
+                logger.warning(
+                    "Controller actor id unavailable; cannot register placement group."
+                )
+                return
+
+            ray.get(
+                self._cleaner.register_controller_and_placement_group.remote(
+                    self._controller_actor_id, placement_group
+                )
+            )
+
+            self._cleaner.start_monitoring.remote()
+
+            logger.debug(
+                f"Registered placement group {placement_group.id} with PlacementGroupCleaner."
+            )
+        except Exception as e:
+            logger.warning(
+                f"Failed to register placement group with cleaner: {e}. "
+                "Placement group may not be cleaned up if controller dies ungracefully."
+            )
+
+    def before_worker_group_shutdown(self, worker_group: "WorkerGroup"):
+        """Stop monitoring the worker group's placement group when it shuts down.
+
+        This is called when a worker group shuts down (e.g., during restart/resize).
+        We stop monitoring the old placement group so that when a new worker group
+        starts, we can start monitoring the new placement group.
+        """
+        if not self._cleaner:
+            return
+
+        try:
+            ray.get(self._cleaner.stop_monitoring.remote(), timeout=2.0)
+            logger.debug("Stopped monitoring placement group for worker group shutdown")
+        except RayActorError:
+            logger.debug(
+                "PlacementGroupCleaner exited before stop_monitoring completed; ignoring."
+            )
+        except Exception:
+            logger.exception("Failed to stop monitoring placement group gracefully.")
```

Comment:
```suggestion
        except Exception:
            logger.exception("Failed to stop placement group gracefully.")
```

don't need to handle actor error separately from other exceptions in this case